src/scripts/main.js

```diff
@@ -1,3 +1,50 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const headers = table.querySelectorAll('th');
+  const tbody = table.querySelector('tbody');
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      sortTableByColumn(index);
+    });
+  });
+
+  function sortTableByColumn(columnIndex) {
+    const rows = Array.from(tbody.querySelectorAll('tr'));
```

Comment:
Consider checking if `tbody` has any rows before attempting to sort. If there are no rows, sorting is unnecessary.